apps/src/code-studio/pd/application_dashboard/principal_approval_buttons.jsx

```diff
@@ -86,17 +106,34 @@ export default class PrincipalApprovalButtons extends React.Component {
       return <Spinner size="small" />;
     }
 
+    const buttonOnClick = this.props.showResendEmailButton
+      ? this.handleResendEmailClick
+      : this.handleSendEmailClick;
+    const buttonText = this.props.showResendEmailButton
+      ? 'Resend request'
+      : 'Send email';
+
     return (
-      <Button
-        bsSize="xsmall"
-        target="_blank"
-        onClick={this.handleSendEmailClick}
-        style={styles.button}
-        // This button is disabled if the other action is pending (which will be rendered as a spinner)
-        disabled={!!this.state.notRequiredRequest}
-      >
-        Send email
-      </Button>
+      <div>
+        <Button
+          bsSize="xsmall"
+          target="_blank"
+          onClick={buttonOnClick}
+          style={styles.button}
+          // This button is disabled if the other action is pending (which will be rendered as a spinner)
+          disabled={!!this.state.notRequiredRequest}
+        >
+          {buttonText}
+        </Button>
+        <ConfirmationDialog
+          show={this.state.showResendEmailConfirmation}
+          onOk={this.handleResendEmailConfirmed}
+          onCancel={this.handleResendEmailCancel}
+          headerText="Resend"
+          bodyText="This will resend an email to this applicant’s principal with a link to the recommendation form. Proceed?"
+          okText="Resend"
+        />
```

Comment:
Probably fine, but might want to spot-check for issues with having a `ConfirmationDialog` component for every single row.

Reply:
Yeah, it would be nice to have a shared confirmation but this seemed lower impact on the whole scene.  I've exercised it with three and not had any issues.  I did try only including it in the DOM when `this.state.showResendEmailConfirmation` but then we lose the nice transition animation when closing it.